tests/foreman/api/test_host.py

```diff
@@ -610,105 +762,88 @@ def test_positive_delete(self):
-    def test_positive_update_domain(self):
-        """Update a host with a new domain
+    def test_positive_create_and_update_domain(self, module_org, module_location):
+        """Create and update a host with a domain
 
         :id: 8ca9f67c-4c11-40f9-b434-4f200bad000f
 
-        :expectedresults: A host is updated with a new domain
+        :expectedresults: A host is created and updated with expected domain
 
         :CaseLevel: Integration
         """
-        host = entities.Host().create()
+        domain = entities.Domain(organization=[module_org], location=[module_location]).create()
+        host = entities.Host(
+            organization=module_org, location=module_location, domain=domain
+        ).create()
+        assert host.domain.read().name == domain.name
+
         new_domain = entities.Domain(
-            location=[host.location], organization=[host.organization]
+            organization=[module_org], location=[module_location]
         ).create()
         host.domain = new_domain
         host = host.update(['domain'])
-        self.assertEqual(host.domain.read().name, new_domain.name)
+        assert host.domain.read().name == new_domain.name
 
     @tier2
-    def test_positive_update_env(self):
-        """Update a host with a new environment
+    def test_positive_create_and_update_env(self, module_org, module_location):
+        """Create and update a host with an environment
 
         :id: 87a08dbf-fd4c-4b6c-bf73-98ab70756fc6
 
-        :expectedresults: A host is updated with a new environment
+        :expectedresults: A host is created and updated with expected environment
 
         :CaseLevel: Integration
         """
-        host = entities.Host().create()
+        env = entities.Environment(organization=[module_org], location=[module_location],).create()
+        host = entities.Host(
+            organization=module_org, location=module_location, environment=env
+        ).create()
+        assert host.environment.read().name == env.name
+
         new_env = entities.Environment(
-            location=[host.location], organization=[host.organization]
+            organization=[host.organization], location=[host.location]
         ).create()
         host.environment = new_env
         host = host.update(['environment'])
-        self.assertEqual(host.environment.read().name, new_env.name)
+        assert host.environment.read().name == new_env.name
 
     @tier2
-    def test_positive_update_arch(self):
-        """Update a host with a new architecture
+    def test_positive_create_and_update_arch(self):
+        """Create and update a host with an architecture
 
         :id: 5f190b14-e6db-46e1-8cd1-e94e048e6a77
 
-        :expectedresults: A host is updated with a new architecture
+        :expectedresults: A host is created and updated with expected architecture
 
         :CaseLevel: Integration
         """
-        host = entities.Host().create()
+        arch = entities.Architecture().create()
```

Comment:
module_arch should be available in global fixtures

Reply:
DONE.